management/actions_test.go

```diff
@@ -53,175 +35,241 @@ func TestActions(t *testing.T) {
+
+	deployedVersion, err := m.Action.Deploy(action.GetID())
+	require.NoError(t, err)
+
+	actualVersions, err := m.Action.Versions(action.GetID())
+
+	assert.NoError(t, err)
+	assert.Equal(t, deployedVersion.GetID(), actualVersions.Versions[0].GetID())
+}
+
+func TestActionManager_Bindings(t *testing.T) {
+	action := givenAnAction(t)
+	ensureActionBuilt(t, action.GetID())
+
+	_, err := m.Action.Deploy(action.GetID())
+	require.NoError(t, err)
+
+	emptyBinding := make([]*ActionBinding, 0)
+	err = m.Action.UpdateBindings(ActionTriggerPostLogin, emptyBinding)
+	assert.NoError(t, err)
+
+	binding := []*ActionBinding{
+		{
+			Ref: &ActionBindingReference{
+				Type:  auth0.String(ActionBindingReferenceByName),
+				Value: action.Name,
 			},
-		}
-		err = m.Action.UpdateBindings(ActionTriggerPostLogin, b)
-		if err != nil {
-			t.Fatal(err)
-		}
-		t.Logf("%v\n", b)
-	})
+			DisplayName: auth0.String("My test action Binding"),
+		},
+	}
 
-	t.Run("Bindings", func(t *testing.T) {
-		bl, err := m.Action.Bindings(ActionTriggerPostLogin)
-		if err != nil {
-			t.Fatal(err)
-		}
-		t.Logf("%v\n", bl)
+	err = m.Action.UpdateBindings(ActionTriggerPostLogin, binding)
+	assert.NoError(t, err)
+
+	bindingList, err := m.Action.Bindings(ActionTriggerPostLogin)
+
+	assert.NoError(t, err)
+	assert.Len(t, bindingList.Bindings, 1)
+
+	t.Cleanup(func() {
+		err = m.Action.UpdateBindings(ActionTriggerPostLogin, emptyBinding)
+		assert.NoError(t, err)
+		cleanupAction(t, action.GetID())
 	})
+}
 
-	t.Run("Test", func(t *testing.T) {
-		p := &ActionTestPayload{
-			"event": ActionTestPayload{
-				"user": ActionTestPayload{
-					"email":         "j+smith@example.com",
-					"emailVerified": true,
-					"id":            "auth0|5f7c8ec7c33c6c004bbafe82",
-				},
+func TestActionManager_Test(t *testing.T) {
+	action := givenAnAction(t)
+	defer cleanupAction(t, action.GetID())
+	ensureActionBuilt(t, action.GetID())
+
+	test := &ActionTestPayload{
+		"event": ActionTestPayload{
+			"user": ActionTestPayload{
+				"email":         "j+smith@example.com",
+				"emailVerified": true,
+				"id":            "auth0|5f7c8ec7c33c6c004bbafe82",
 			},
-		}
-		err = m.Action.Test(r.GetID(), p)
-		if err != nil {
-			t.Fatal(err)
-		}
-		t.Logf("%v\n", p)
-	})
+		},
+	}
+	err := m.Action.Test(action.GetID(), test)
+	assert.NoError(t, err)
+}
 
-	t.Run("Execution", func(t *testing.T) {
-		_, err := m.Action.Execution("M9IqRp9wQLaYNrSwz6YPTTIwMjEwNDA0")
-		if err != nil {
-			mgmtError, _ := err.(*managementError)
-			if mgmtError.StatusCode != 404 {
-				t.Fatal(err)
-			}
-			// Expect a 404 as we can't get execution ID via API
-			t.Log(err)
-			return
-		}
-		t.Fatal(errors.New("read execution unexpectedly succeeded"))
-	})
+func TestActionManager_Execution(t *testing.T) {
+	_, err := m.Action.Execution("M9IqRp9wQLaYNrSwz6YPTTIwMjEwNDA0")
+	// Expect a 404 as we can't get execution ID via API
+	assert.EqualError(t, err, "404 Not Found: That execution does not exist.")
```

Comment:
Same, can we assert by the status code (404) instead?

Reply:
Yep, good idea 👍🏻  [954a359](https://github.com/auth0/go-auth0/pull/49/commits/954a3599e0b5ea0f018239b53bd9af5f261bac8f)